### ISSUE TITLE
Test

### ISSUE THREAD

**Issue by Zakobian** [Zakobian/CompFrm#1]
The OperatorAsModule doesn't seem to be able to propogate backwards after having done it once. An example is the following (Trying to calculate gradient to do gradient descent), using an MNIST image: 
`trafo = transforms.Compose([transforms.ToTensor()])
dset_train = datasets.MNIST('./', train=True, download=True, transform=trafo)
train_loader = torch.utils.data.DataLoader(dset_train, batch_size=50, shuffle=True)
space = odl.uniform_discr([-14, -14], [14, 14], [28, 28], dtype='float32')
geometry = odl.tomo.parallel_beam_geometry(space, num_angles=5)
fwd_op = odl.tomo.RayTransform(space, geometry)
fbp_op = odl.tomo.fbp_op(fwd_op)
fwd_op_mod = OperatorAsModule(fwd_op)
fbp_op_mod = OperatorAsModule(fbp_op)
def generate_data(images):
    torch.manual_seed(123)
    data = fwd_op_mod(images)
    data += Variable(torch.randn(data.shape)).type_as(data)
    return data
for i, (images, _) in enumerate(test_loader):
    if i == 1:
        break

guess=fbp_op_mod(generate_data(test_images))[0:1,0:1].clone()
truth=Variable(images)[0:1,0:1]`
